airbyte-integrations/connectors/source-mysql/src/main/java/io/airbyte/integrations/source/mysql/MySqlSource.java

```diff
@@ -338,6 +349,54 @@ public List<AutoCloseableIterator<AirbyteMessage>> getIncrementalIterators(final
       LOGGER.info("Using PK + CDC");
       return MySqlInitialReadUtil.getCdcReadIterators(database, catalog, tableNameToTable, stateManager, emittedAt, getQuoteString());
     } else {
+      if (isAnyStreamIncrementalSyncMode(catalog)) {
```

Comment:
Do we want to gate this via `MySqlFeatureFlags`?

Reply:
Gated!